src/supertux/sector_base.cpp

```diff
@@ -33,6 +31,29 @@ Sector::run_script(const std::string& script, const std::string& sourcename)
   m_squirrel_environment->run_script(script, sourcename);
 }
 
+void
+Sector::run_script(const std::string& script, const std::string& sourcename,
+                   const GameObject& object, std::map<std::string, std::string> triggers)
+{
+  if (!object.get_name().empty())
+  {
+    // Delete any existing trigger references
+    m_squirrel_environment->modify_table([&object](SquirrelVM& vm) {
+        try
+        {
+          vm.get_table_entry("triggers");
+          vm.delete_table_entry(object.get_name().c_str());
+        }
+        catch (...) {}
+      });
+
+    for (auto it = triggers.begin(); it != triggers.end(); it++)
+      m_squirrel_environment->create_reference(it->first, it->second, "triggers", object.get_name());
```

Comment:
Since we are using C++17 you can use structured binding:
```cpp
for (const auto& [name, class_name] : triggers)
  m_squirrel_environment->create_reference(name, class_name, "triggers", object.get_name());
```